src/convenience/session.ts

```diff
@@ -85,6 +85,14 @@ export interface StorageAdapter<T> {
      * Deletes a value for the given key from the storage.
      */
     delete: (key: string) => MaybePromise<void>;
+    /**
+     * Checks whether a key exists in the storage
+     */
+    has?: (key: string) => MaybePromise<boolean>;
+    /**
+     * Lists all keys
+     */
+    readAll?: () => Iterable<T> | AsyncIterable<T>;
```

Comment:
It does not list all the keys. But perhaps we should consider doing this instead. I think only retrieving values without the corresponding keys is rarely useful, so I see three alternatives here:
1. Provide `readAllKeys` only
2. Provide `readAllKeys` and `readAllEntries` which returns `[key, value]` pairs
3. Provide `readAllKeys` and `readAllEntries` and `readAllValues`

Also, we have to think about `readAll` from the current memory sessions. Should we align the name in the adapter for 2.0 (deprecate until then) or rather align the name in the interface to `readAll`?

Reply:
> It does not list all the keys. But perhaps we should consider doing this instead. I think only retrieving values without the corresponding keys is rarely useful, so I see three alternatives here:

You're right.
I haven't thought about the implementation of the broadcast plugin enough to be able to tell if we'll need keys, entries, or values, so I'd say we go with 3. to be future proof.

> Should we align the name in the adapter for 2.0 (deprecate until then)

I'm assuming by this you mean we add the three new methods and have `readAll` call `readValues` (current behavior), then remove it on 2.0, right? If so, I think this is the way to go

Reply:
Yes, sounds good then!